### PR TITLE
Create new ButtonAction component

### PR DESCRIPTION
**Pull request by cdedreuille**
<details>
  <summary>📦 Published PR as canary version: <code>7.14.1-canary.412.ca873c8.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/design-system@7.14.1-canary.412.ca873c8.0
  # or 
  yarn add @storybook/design-system@7.14.1-canary.412.ca873c8.0
  ```
</details>

### CODE REVIEW

File: src/components/ButtonAction.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  display: flex;
+  gap: 6px;
+  align-items: center;
+  overflow: hidden;
+  padding: 7px;
+  position: relative;
+  text-align: center;
+  text-decoration: none;
+  transition: all 150ms ease-out;
+  transform: translate3d(0, 0, 0);
```
Comment (winkerVSbecks) [inline src/components/ButtonAction.tsx:37]: Doesn't look like this button uses the `transform` animation. You could probably delete lines 37-38

File: src/components/ButtonAction.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const meta: Meta<typeof ButtonAction> = {
+  title: 'ButtonAction',
+  component: ButtonAction,
+  tags: ['autodocs'],
+};
+
+export default meta;
+type Story = StoryObj<typeof ButtonAction>;
+
+export const Default: Story = {
```
Comment (winkerVSbecks) [inline src/components/ButtonAction.stories.tsx:14]: I would add stories for all permutations:
* icon only 
* icon + label
* active
* loading
* with tooltip

File: src/components/ButtonAction.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          isLoading={isLoading}
+          loadingText={loadingText}
+          {...rest}
+        >
+          {children}
+        </InsideButtonAction>
+      </WithTooltip>
+    );
+  return (
+    <InsideButtonAction
```
Comment (winkerVSbecks) [inline src/components/ButtonAction.tsx:87]: I like this `Inside` terminology

File: src/components/ButtonAction.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      <div style={{ position: 'relative', width: 14, height: 14 }}>
+        <Spinner inForm />
+      </div>
+    )}
+    {children && !isLoading && children}
+    {isLoading && loadingText}
+  </StyledButton>
+);
+
+ButtonAction.displayName = 'ButtonAction';
```
Comment (winkerVSbecks) [inline src/components/ButtonAction.tsx:120]: You don't need this. It's only required if you use `React.forwardRef`

File: src/components/ButtonAction.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  icon,
+  isActive = false,
+  isLoading = false,
+  loadingText = null,
+  tooltip,
+  ...rest
+}) => {
+  if (tooltip)
+    return (
+      <WithTooltip tooltip={<TooltipNote note="Coco" />} hasChrome={false}>
```
Comment (winkerVSbecks) [inline src/components/ButtonAction.tsx:74]: should the `note` be hard coded?

**Review by winkerVSbecks — CHANGES_REQUESTED** [review]

File: src/components/ButtonAction.tsx (hunk 5 above)
Comment (cdedreuille) [reply src/components/ButtonAction.tsx:74]: Oh god ... it should come from the props. I forgot to replace it.

File: src/components/ButtonAction.stories.tsx (hunk 2 above)
Comment (cdedreuille) [reply src/components/ButtonAction.stories.tsx:14]: I added those.

File: src/components/ButtonAction.tsx (hunk 4 above)
Comment (cdedreuille) [reply src/components/ButtonAction.tsx]: Yes. Left over. I removed it.

**Review by winkerVSbecks — APPROVED** [review]

**Review by MichaelArestad — APPROVED** [review]

**Pull request merged**